frontend-react/src/__mocks__/DBMock.test.ts

```diff
@@ -0,0 +1,239 @@
+import { SetupServer, setupServer } from "msw/node";
+
+import { RSOrganizationSettings } from "../config/endpoints/settings";
+import { createDbMock } from "../utils/MSWData/CreateDbMock";
+import {
+    RelativePathMap,
+    ParentResolverMap,
+    factoryRestHandlerTupleToMap,
+    RestHandlerMethods,
+} from "../utils/MSWData/EnhancedFactory";
+
+import { createModels } from "./DBMock";
+
+let { db, reset } = createDbMock(createModels);
+
+describe("DBMock", () => {
+    beforeEach(() => {
+        reset();
+    });
+
+    describe("createMany", () => {
+        const createManyScenarios: {
+            label: string;
+            args: [number, Partial<RSOrganizationSettings>[] | undefined];
+        }[] = [
+            {
+                label: "predictably random",
+                args: [2, undefined],
+            },
+            {
+                label: "all with initial values",
+                args: [2, [{ name: "test1" }, { name: "test2" }]],
+            },
+            {
+                label: "some with initial values",
+                args: [2, [{ stateCode: "LA" }]],
+            },
+        ];
+
+        test.each(createManyScenarios)("$label", ({ args }) => {
+            const [num, initial = []] = args;
+            db.organizationSettings.createMany(...args);
+            expect(db.organizationSettings.count()).toStrictEqual(num);
+            const all = db.organizationSettings.getAll();
+            for (const [i, init] of initial.entries()) {
+                for (const [k, v] of Object.entries(init)) {
+                    // check initialized value
+                    expect(
+                        all[i][k as keyof (typeof all)[number]]
+                    ).toStrictEqual(v);
+                }
+            }
+            if (initial.length === 0) {
+                // check is random
+                expect(all[0]).not.toStrictEqual(all[1]);
+                reset();
+                db.organizationSettings.createMany(2);
+                // check is predictably random
+                expect(db.organizationSettings.getAll()).toStrictEqual(all);
+            }
+        });
+    });
+
+    describe("json serialized return methods", () => {
```

Comment:
I don't know if there's a better way of doing this since we're randomizing the records, but it kiiiiiiind of feels like we're reimplementing the source code logic in these tests.  I wonder if it'd be better to initialize the database with records we can reference instead like:

```tsx
const records = [{
    name: "first organization"
}, {
    name: "second organization"
}];

db.organizationCreateMany(records.length, records);
```

Then we could run expectations accordingly on `records` instead of needing to call `getAll` directly 🤔 Though I know that the randomized properties could cause some issues there so this isn't a hard blocker in my mind.